metashare/consumer_utils.py

```diff
@@ -15,7 +15,9 @@ def message_to_hash(message):
 async def get_set_message_semaphore(channel_layer, message):
     msg_hash = message_to_hash(message)
     async with channel_layer.connection(0) as connection:
-        return await connection.setnx(msg_hash, 1)
+        ret = await connection.setnx(msg_hash, 1)
+        await connection.expire(msg_hash, 2)
+        return ret
```

Comment:
You can do this with a single call to redis: https://github.com/SFDO-Tooling/MetaDeploy/blob/master/metadeploy/consumer_utils.py#L21 (we used 5 seconds there, but I think that was arbitrary)